affinity/cpp-20/d0796r1.md

```diff
@@ -136,25 +142,54 @@ There are some additional challenges which we have been investigating but are no
 
     /* Execution context */
 
-    struct execution_context {
+    class execution_context {
+     public:
 
       using executor_type = __unspecfied__;
 
-      template <typename ExecutionResource>
-      execution_context(ExecutionResource &&execResource);
+      execution_context(const execution_resource &);
 
       ~execution_context();
 
       const execution_resource &resource() const noexcept;
 
       executor_type executor() noexcept;
 
+      pmr::memory_resource *memory_resource() noexcept;
```

Comment:
So originally made this by pointer as the `pmr::polymorphic_allocator` expects a raw pointer to a memory resource, but I agree that this is not ideal. I think this should be returned by reference instead since if it's returned by value you'll end up with a copy of the memory resource.

You make a good point about the `noexcept`, I will remove this. I'll also put a throws clause to state that an exception will be thrown if the execution context cannot allocate memory. And i'll add a similar requirement for the `executor` member function.

I'll add an `allocator` member function too.